File: common/src/test/java/dev/number6/slack/FakeHttpAdaptor.java
package dev.number6.slack;

import dev.number6.slack.port.HttpPort;
import com.amazonaws.services.lambda.runtime.LambdaLogger;

public class FakeHttpAdaptor implements HttpPort {

    private String response = "response";

    @Override
    public CallResponse get(String url, LambdaLogger logger) {
        return new CallResponse(response);
    }

    @Override
    public CallResponse post(String url, String body, LambdaLogger logger) {
        return new CallResponse(response);
    }

    public void setResponse(String response) {
        this.response = response;
    }
}

File: common/src/test/java/dev/number6/slack/FakeHttpAdaptor.kt
package dev.number6.slack

import com.amazonaws.services.lambda.runtime.LambdaLogger
import dev.number6.slack.port.HttpPort

class FakeHttpAdaptor : HttpPort {
    private var response = "response"
    override fun get(url: String, logger: LambdaLogger): CallResponse {
        return CallResponse(response)
    }

    override fun post(url: String, body: String, logger: LambdaLogger): CallResponse {
        return CallResponse(response)
    }

    fun setResponse(response: String) {
        this.response = response
    }
}